File: algorithm/array_and_matrices/217.cpp
#include <iostream>
#include <map>
using namespace std;

class Solution
{
public:
    bool containsDuplicate(vector<int> &nums)
    {
        map<int, bool> myMap;
        for (const auto &n : nums)
        {
            if (myMap.find(n) == myMap.end())
            {
                myMap[n] = true;
            }
            else
            {
                return true;
            }
        }
        {
        }
        return false;
    }
};

int main()
{
    Solution sol;
    vector<int> nums = {1, 2, 3, 1};
    cout << boolalpha << sol.containsDuplicate(nums);
}

File: algorithm/array_and_matrices/217.py
from typing import List
import numpy as np

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        dictionary = {}
        for i in nums:
            if i in dictionary:
                return True
            else:
                dictionary[i] = 1
        return False


if __name__ == "__main__":
    print(Solution().containsDuplicate([1,1,1,3,3,4,3,2,4,2]))